olp-cpp-sdk-core/src/cache/DefaultCache.cpp

```diff
@@ -281,7 +288,19 @@ DefaultCache::StorageOpenResult DefaultCache::SetupStorage() {
   if (settings_.max_memory_cache_size > 0) {
     memory_cache_.reset(new InMemoryCache(settings_.max_memory_cache_size));
   }
-  if (settings_.disk_path) {
+
+  // For backwards compatibility.
+  // If the user provided old disk_path instead of a new one, use it.
+  if (!settings_.disk_path_mutable) {
+    PORTING_PUSH_WARNINGS()
+    PORTING_CLANG_GCC_DISABLE_WARNING("-Wdeprecated-declarations")
+    PORTING_MSVC_DISABLE_WARNINGS(4996)
+    // Direct usage of deprecated field
```

Comment:
Same question about the usage of comments.

Reply:
removed